src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -224,6 +211,72 @@ impl LeaderBlockCommitOp {
             return Err(op_error::ParseError);
         }
 
+        let mut commit_outs = vec![];
+        let mut pox_fee = None;
+        let mut burn_fee = None;
+
+        for (ix, output) in outputs.into_iter().enumerate() {
+            // only look at the first OUTPUTS_PER_COMMIT outputs
+            //   or until first _burn_ output
+            if ix >= OUTPUTS_PER_COMMIT {
+                break;
+            }
+            if output.address.is_burn() {
+                burn_fee.replace(output.amount);
+                break;
+            } else {
+                // all pox outputs must have the same fee
+                if let Some(pox_fee) = pox_fee {
+                    if output.amount != pox_fee {
+                        warn!("Invalid commit tx: different output amounts for different PoX reward addresses");
+                        return Err(op_error::ParseError);
+                    }
+                } else {
+                    pox_fee.replace(output.amount);
+                }
+                commit_outs.push(output.address);
+            }
+        }
+
+        // EITHER there was an amount burned _or_ there were OUTPUTS_PER_COMMIT pox outputs
+        if burn_fee.is_none() && commit_outs.len() != OUTPUTS_PER_COMMIT {
+            warn!("Invalid commit tx: if fewer than {} PoX addresses are committed to, remainder must be burnt", OUTPUTS_PER_COMMIT);
+            return Err(op_error::ParseError);
+        }
+
+        // compute the total amount transfered/burned, and check that the burn amount
+        //   is expected given the amount transfered.
+        let burn_fee = 
+            match (burn_fee, pox_fee) {
+                (Some(burned_amount), Some(pox_amount)) => {
+                    // burned amount must be equal to the "missing"
+                    //   PoX slots
+                    let expected_burn_amount = pox_amount.checked_mul((OUTPUTS_PER_COMMIT - commit_outs.len()) as u64)
+                        .ok_or_else(|| op_error::ParseError)?;
+                    if expected_burn_amount != burned_amount {
+                        warn!("Invalid commit tx: burned output different from PoX reward output");
+                        return Err(op_error::ParseError);
+                    }
+                    pox_amount.checked_mul(OUTPUTS_PER_COMMIT as u64)
+                        .ok_or_else(|| op_error::ParseError)?
+                },
+                (Some(burned_amount), None) => {
+                    burned_amount
+                },
+                (None, Some(pox_amount)) => {
+                    pox_amount.checked_mul(OUTPUTS_PER_COMMIT as u64)
+                        .ok_or_else(|| op_error::ParseError)?
+                },
+                (None, None) => {
+                    unreachable!("A 0-len output should have already errored");
+                },
+            };
+
+        if burn_fee <= 0 {
```

Comment:
I think that `burn_fee` is a u64.

Reply:
It is, but you can never be too sure about those negative integers :)

Update to just do an equality check in 4a6a38a30